pkg/network/mycelium/mycelium.go

```diff
@@ -0,0 +1,235 @@
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/zos/pkg/network/namespace"
+	"github.com/threefoldtech/zos/pkg/zinit"
+)
+
+const (
+	tunName      = "my0"
+	myBin        = "mycelium"
+	zinitService = "mycelium"
+	confPath     = "/tmp/mycelium_priv_key.bin"
+	MyListenTCP  = 9651
+)
+
+// MyServer represent a mycelium server
+type MyServer struct {
+	cfg *NodeConfig
+	ns  string
+}
+
+type MyceliumInspection struct {
+	PublicKey string `json:"publicKey"`
+	Address   net.IP `json:"address"`
+}
+
+type Subnet [8]byte
+
+// NewMyServer create a new mycelium Server
+func NewMyServer(cfg *NodeConfig) *MyServer {
+	return &MyServer{
+		cfg: cfg,
+	}
+}
+
+func (s *MyServer) Restart(z *zinit.Client) error {
+	return z.Kill(zinitService, zinit.SIGTERM)
+}
+
+func (s *MyServer) Reload(z *zinit.Client) error {
+	return z.Kill(zinitService, zinit.SIGHUP)
+}
+
+// Start creates a mycelium zinit service and starts it
+func (s *MyServer) Ensure(z *zinit.Client, ns string) error {
+	if !namespace.Exists(ns) {
+		return fmt.Errorf("invalid namespace '%s'", ns)
+	}
+	s.ns = ns
+
+	if err := writeKey(confPath, s.cfg.PrivateKey); err != nil {
+		return err
+	}
+
+	// service found.
+	// better if we just stop, forget and start over to make
+	// sure we using the right exec params
+	if _, err := z.Status(zinitService); err == nil {
+		// not here we need to stop it
+		if err := z.StopWait(5*time.Second, zinitService); err != nil && !errors.Is(err, zinit.ErrUnknownService) {
+			return errors.Wrap(err, "failed to stop mycelium service")
+		}
+		if err := z.Forget(zinitService); err != nil && !errors.Is(err, zinit.ErrUnknownService) {
+			return errors.Wrap(err, "failed to forget mycelium service")
+		}
+	}
+
+	bin, err := exec.LookPath(myBin)
+	if err != nil {
+		return err
+	}
+
+	cmd := `sh -c '
+    exec ip netns exec %s %s --key-file %s --tun-name %s --peers %s
+    '`
+
+	err = zinit.AddService(zinitService, zinit.InitService{
+		Exec: fmt.Sprintf(cmd, ns, bin, confPath, s.cfg.TunName, strings.Join(s.cfg.Peers, " ")),
+		After: []string{
+			"node-ready",
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	if err := z.Monitor(zinitService); err != nil && !errors.Is(err, zinit.ErrAlreadyMonitored) {
+		return err
+	}
+
+	return z.StartWait(time.Second*20, zinitService)
+}
+
+// NodeID returns mycelium node ID of s
+func (s *MyServer) NodeID() (ed25519.PublicKey, error) {
+	if s.cfg.PublicKey == "" {
+		panic("EncryptionPublicKey empty")
+	}
+
+	return hex.DecodeString(s.cfg.PublicKey)
+}
+
+func (s *MyServer) InspectMycelium() (inspection MyceliumInspection, err error) {
+	// we check if the file exists before we do inspect because mycelium will create a random seed
+	// file if file does not exist
+	_, err = os.Stat(confPath)
+	if err != nil {
+		return inspection, err
+	}
+
+	bin, err := exec.LookPath(myBin)
+	if err != nil {
+		return inspection, err
+	}
+
+	cmd := fmt.Sprintf(`exec ip netns exec %s %s inspect --json --key-file %s `, s.ns, bin, confPath)
```

Comment:
Any idea why your command is a `sh` then u do exec and not the command directly!
I mean why not do 
```
os.Command("ip", "netns", "exec", s.ns,  bin, "inspect", "--json", "--key-file", confPath)
```
?!